File: javascript/test/data-structures/linked-list/nodeSpec.js
require('../../test_helper')
describe('Node', function () {
  var data = "sweet data",
      node;
  beforeEach(function () {
    node = new Node(data);
  });
  describe('#data', function () {
    it('is initialized to the passed in value', function () {
      expect(node.data).to.eq(data);
    });
    it('raises if data is undefined', function () {
      expect(function() { new Node }).to.throw(Error);
    });
    it('we are able to set it', function () {
      var anoterData = "another data";
      node.data = anoterData;
      expect(node.data).to.eq(anoterData);
    });
  });
  describe('#next', function () {
    it('is initialized to nil', function () {
      expect(node.next).be.an('undefined');
    });
  });
  describe('#prev', function () {
    it('is initialized to nil', function () {
      expect(node.prev).be.an('undefined');
    });
  });
});

File: ruby/spec/data-structures/linked-list/node_spec.rb
describe Node do
  let(:data) { "sweet" }
  let(:next_node) { "awesome" }
  let(:node) { Node.new data }
  context "#data" do
    it "is initialized to the given data" do
      expect(node.data).to eq data
    end
  end
  context "#next" do
    it "defaults to nil" do
      expect(node.next).to be_nil
    end
  end
  context "#next=" do
    it "sets the next" do
      node.next = next_node
      expect(node.next).to eq next_node
    end
  end
end
